packages/manager/modules/user-contracts/src/user-contracts.service.js

```diff
@@ -0,0 +1,67 @@
+export default class {
+  /* @ngInject */
+  constructor($q, OvhHttp) {
+    this.$q = $q;
+    this.OvhHttp = OvhHttp;
+  }
+
+  getAgreementsToValidate(predicate) {
+    return this.getTodoAgreements().then((contracts) =>
+      contracts.filter(predicate),
+    );
+  }
+
+  acceptAgreements(agreements) {
+    const promises = agreements.map((agreement) =>
+      this.acceptAgreement(agreement),
+    );
+    return this.$q.all(promises);
+  }
+
+  acceptAgreement(agreement) {
+    return this.OvhHttp.post(`/me/agreements/${agreement.id}/accept`, {
+      rootPath: 'apiv6',
+    });
+  }
+
+  getTodoAgreements() {
+    return this.OvhHttp.get('/me/agreements', {
```

Comment:
`$http.get('/sws/agreements', {
          params: {
            count: 0,
            offset: 0,
            agreed: 'todo',
          },
          serviceType: 'aapi',
        })`
The above code can help avoid the additional call to get the agreement details